src/SIM/SIMoutput.C

```diff
@@ -230,34 +241,77 @@ bool SIMoutput::parseOutputTag (const tinyxml2::XMLElement* elem)
       free(sval);
     }
 
+    bool elementGrid = false;
+    utl::getAttribute(grid,"elements",elementGrid);
+
     if (newGroup)
       myPoints.push_back({"",{}});
     newGroup = false;
 
-    for (int k = 0; k < npt[2]; k++)
+    if (elementGrid)
     {
-      double zeta = npt[2] > 1 ? double(k)/double(npt[2]-1) : 0.0;
-      thePoint.u[2] = u0[2]*(1.0-zeta) + u1[2]*zeta;
-      for (int j = 0; j < npt[1]; j++)
+      thePoint.npar = 0;
+      myPoints.back().second.push_back(thePoint);
+    }
+    else if (cartesianGrid)
+    {
+      thePoint.inod = -npt[0];
+      Vec3 dX = X1 - X0;
+      for (int d = 0; d < 3; d++)
+        if (center && npt[d] > 1)
+        {
+          dX[d] /= double(npt[d]);
+          X0[d] += 0.5*dX[d];
+        }
+        else if (npt[d] > 2)
+          dX[d] /= double(npt[d]-1);
```

Comment:
okay, removed this cast and the other places where it will happen automatically in any case (when one of the operands is a double). Need to retain when both are int's to avoid integer division.